File: Sources/iStream/UI/TargetsSection.swift
#if canImport(SwiftUI)
import SwiftUI

public struct TargetsSection: View {
    @ObservedObject var vm: PipelineFormViewModel

    public init(vm: PipelineFormViewModel) {
        self.vm = vm
    }

    public var body: some View {
        Section {
            HStack {
                Text("Throughput (events/sec)")
                Spacer()
                TextField("50000", value: $vm.throughputEventsPerSec, format: .number)
                    .keyboardType(.numberPad)
                    .multilineTextAlignment(.trailing)
                    .frame(maxWidth: 140)
            }

            VStack(alignment: .leading) {
                HStack {
                    Text("Peak multiplier")
                    Spacer()
                    Text("×\(vm.peakMultiplier, specifier: "%.1f")")
                        .foregroundStyle(.secondary)
                }
                Slider(value: $vm.peakMultiplier, in: 1.0...10.0, step: 0.5)
            }

            HStack {
                Text("Avg event size (bytes)")
                Spacer()
                TextField("512", value: $vm.avgEventSizeBytes, format: .number)
                    .keyboardType(.numberPad)
                    .multilineTextAlignment(.trailing)
                    .frame(maxWidth: 140)
            }

            HStack {
                Text("Latency SLO (ms)")
                Spacer()
                TextField("1000", value: $vm.latencySLOms, format: .number)
                    .keyboardType(.numberPad)
                    .multilineTextAlignment(.trailing)
                    .frame(maxWidth: 140)
            }
        } header: {
            Text("Throughput & Latency Targets")
        } footer: {
            Text("Peak multiplier scales base throughput for traffic spikes. Latency SLO bounds checkpointing and watermark intervals.")
        }
    }
}
#endif

File: lib/ui/sections/targets_section.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

import '_section_card.dart';

class TargetsSection extends StatelessWidget {
  const TargetsSection({
    super.key,
    required this.throughputEventsPerSec,
    required this.onThroughputChanged,
    required this.peakMultiplier,
    required this.onPeakChanged,
    required this.avgEventSizeBytes,
    required this.onAvgEventSizeChanged,
    required this.latencySLOms,
    required this.onLatencySLOChanged,
  });

  final int throughputEventsPerSec;
  final ValueChanged<int> onThroughputChanged;
  final double peakMultiplier;
  final ValueChanged<double> onPeakChanged;
  final int avgEventSizeBytes;
  final ValueChanged<int> onAvgEventSizeChanged;
  final int latencySLOms;
  final ValueChanged<int> onLatencySLOChanged;

  @override
  Widget build(BuildContext context) {
    return SectionCard(
      title: 'Throughput & Latency Targets',
      footer:
          'Peak multiplier scales base throughput for traffic spikes. Latency SLO bounds checkpointing and watermark intervals.',
      children: [
        _IntField(
          label: 'Throughput (events/sec)',
          value: throughputEventsPerSec,
          onChanged: onThroughputChanged,
        ),
        const SizedBox(height: 12),
        Row(
          children: [
            const Text('Peak multiplier'),
            const Spacer(),
            Text('×${peakMultiplier.toStringAsFixed(1)}'),
          ],
        ),
        Slider(
          value: peakMultiplier,
          min: 1.0,
          max: 10.0,
          divisions: 18,
          label: '×${peakMultiplier.toStringAsFixed(1)}',
          onChanged: onPeakChanged,
        ),
        _IntField(
          label: 'Avg event size (bytes)',
          value: avgEventSizeBytes,
          onChanged: onAvgEventSizeChanged,
        ),
        const SizedBox(height: 12),
        _IntField(
          label: 'Latency SLO (ms)',
          value: latencySLOms,
          onChanged: onLatencySLOChanged,
        ),
      ],
    );
  }
}

class _IntField extends StatelessWidget {
  const _IntField({
    required this.label,
    required this.value,
    required this.onChanged,
  });

  final String label;
  final int value;
  final ValueChanged<int> onChanged;

  @override
  Widget build(BuildContext context) {
    return TextFormField(
      initialValue: value.toString(),
      decoration: InputDecoration(labelText: label),
      keyboardType: TextInputType.number,
      inputFormatters: [FilteringTextInputFormatter.digitsOnly],
      onChanged: (v) => onChanged(int.tryParse(v) ?? 0),
    );
  }
}
